### PR TITLE
feat(engine): Enforce operation limits in engine v1

### PR DESCRIPTION
**Pull request by jakubadamw**
# Description

Add enforcement of operation limits as declared in the SDL to the 1st version of the engine.

# Type of change

- [ ] 💔 Breaking
- [X] 🚀 Feature
- [ ] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

File: engine/crates/engine/src/schema.rs [hunk 1]
```diff
@@ -186,8 +172,10 @@ impl Deref for SchemaEnv {
 #[doc(hidden)]
 pub struct SchemaInner {
     pub(crate) validation_mode: ValidationMode,
-    pub(crate) complexity: Option<usize>,
-    pub(crate) depth: Option<usize>,
+    pub(crate) max_complexity: Option<usize>,
```
Comment (Finistere) [inline engine/crates/engine/src/schema.rs:175]: Could also be in a `OperationLimits` struct with a `Default`

File: engine/crates/engine/src/validation/mod.rs [hunk 2]
```diff
@@ -32,6 +32,12 @@ pub struct ValidationResult {
 
     /// Query depth
     pub depth: usize,
+
+    /// Query height
```
Comment (Finistere) [inline engine/crates/engine/src/validation/mod.rs:36]: Not super obvious what `height` meant before reading the tests to be honest^^

**Review by Finistere — APPROVED** [review]

File: engine/crates/engine/src/schema.rs (hunk 1 above)
Comment (jakubadamw) [reply engine/crates/engine/src/schema.rs:175]: Good idea!

File: engine/crates/engine/src/validation/mod.rs (hunk 2 above)
Comment (jakubadamw) [reply engine/crates/engine/src/validation/mod.rs:36]: Added a comment.

**Pull request merged**